### PR TITLE
Fix config.yaml

### LINKED ISSUE #138: [BUG] config.yaml is not populated with hostvars
For example host.ini

```
[rke2_servers]
master-01 node_labels='["extraLabel0=true"]' node_ip="10.0.0.10" node_name="customName0" bind_address="10.0.0.10" advertise_adress="10.0.0.10" node_external_ip="52.52.52.52" node_taints='["CriticalAddonsOnly=true:NoSchedule"]' cloud_provider_name="aws"
```

The resulting config will be:
```
cat /etc/rancher/rke2/conf.yaml 
cloud-provider-name: aws
node-label:
- extraLabel0=true
node-taint:
- CriticalAddonsOnly=true:NoSchedule
```

Additionally when running with variable `rke2_config['node-labels']` or `rke2_config['nodte-taint']` resulting config.yaml file is even smaller and will contain just node-label/node-taint respectively (plus server and token for nodes)
```
ansible-playbook site.yml -v -l master-01 -e '{ "rke2_config" : { "node-label": ["ansibleLabel=true"] }}'
...
cat /etc/rancher/rke2/config.yaml
node-label:
- ansibleLabel=true
```

### PR DESCRIPTION
**Pull request by marcinkubica**
## What type of PR is this?

_(REQUIRED)_

- [X] bug
- [ ] cleanup
- [ ] documentation
- [ ] feature

## What this PR does / why we need it:
The current logic for building config.yaml doesn't populate values properly. 
To simplify and fix the proposed fix is using the jinia2 template.

## Which issue(s) this PR fixes:

Fixes #138

## Release Notes

```release-note
NONE
```

### CODE REVIEW

File: roles/rke2_common/templates/config.yml.j2 [hunk 1]
```diff
@@ -0,0 +1,28 @@
+{% if advertise_address is defined and advertise_address %}
+advertise-address: {{ addvertise_adress }}
```
Comment (phil-dotchon) [inline roles/rke2_common/templates/config.yml.j2:2]: seems like a typo to me? {{ addvertise_adress }}

Comment (marcinkubica) [reply roles/rke2_common/templates/config.yml.j2]: thanks @phil-dotchon , fixed in https://github.com/rancherfederal/rke2-ansible/pull/139/commits/8397bf5b57e32b90fb602bac42dad8e6df953a87

**Comment by mddamato** [comment]
Thank you for contributing @marcinkubica. Unfortunately, this change does not work as it did previously. I like the approach you took, but we also need to account for other variables that can be passed into the RKE2 config file. The roles were written so you can pass in arbitrary RKE2 configuration. The logic in the config.yml was to allow for the possibility of adding _certain_ variables to both host_vars and group_vars while also being open-ended for the rest of the [config options](https://docs.rke2.io/reference/server_config). 

for example, an RKE2 config can be provided via group_vars like this:
```
rke2_config:
  selinux: false
  node-label:
    - groupvarlabel1=true
    - groupvarlabel2=true
  kube-controller-manager-arg:
    - "tls-min-version=VersionTLS12"
 ```
 
 with your change, the SELinux and controller args are dropped.

**Review by mddamato — CHANGES_REQUESTED** [review]
https://github.com/rancherfederal/rke2-ansible/pull/139#issuecomment-1505193231

**Comment by marcinkubica** [comment]
thanks for the info @mddamato 
I won't be able to continue on this anymore - feel free to drop/develop as needed

**Pull request closed without merging**